confmap/expand.go

```diff
@@ -138,11 +142,26 @@ func toString(strURI string, input any) (string, error) {
 	}
 }
 
-func (mr *Resolver) expandURI(ctx context.Context, uri string) (any, bool, error) {
-	lURI, err := newLocation(uri[2 : len(uri)-1])
-	if err != nil {
-		return nil, false, err
+func (mr *Resolver) expandURI(ctx context.Context, input string) (any, bool, error) {
+	// strip ${ and }
+	uri := input[2 : len(input)-1]
+
+	var lURI location
+	var err error
+	// If no ':' then we have no scheme, check if there is a default scheme configured
+	if !strings.Contains(uri, ":") {
+		// If we have no default scheme there is nothing that can be expanded, return the original input unchanged
+		if mr.defaultScheme == "" {
+			return input, false, nil
+		}
+		lURI = location{scheme: mr.defaultScheme, opaqueValue: uri}
+	} else {
```

Comment:
There is a behavior change here, before having `${TEST_ENV}${file:foo.txt}` would expend the file always, with the current change you will find the first "uri" and will return that nothing changed and will not call again to find `${file:foo.txt}`, can you add a test for this?

I think in findUri you need to pass one option to say "skipNoSchemaUri" if  `mr.defaultScheme == ""`. Then you change it to only check for `:` if `skipNoSchemaUri == true`

Reply:
I think you're right good catch: https://github.com/open-telemetry/opentelemetry-collector/pull/10246